_resources/port1.0/group/app-1.0.tcl

```diff
@@ -218,18 +218,24 @@ platform macosx {
                 return -code error "app.executable ${app.executable} should not start with \${destroot}"
             }
 
-            # If app.executable is in the destroot, link to it.
+            # If app.executable is in the destroot, write a script to launch it.
             if {[file exists ${destroot}[app._resolve_symlink ${executable} ${destroot}]]} {
-                ln -s ${executable} ${destroot}${applications_dir}/${app.name}.app/Contents/MacOS/${app.name}
+                set launch_script [open ${destroot}${applications_dir}/${app.name}.app/Contents/MacOS/${app.name} w]
+                puts ${launch_script} "#!/bin/bash\n${executable}\n"
+                close ${launch_script}
+                system -W ${worksrcpath} "/bin/chmod 755 ${destroot}${applications_dir}/${app.name}.app/Contents/MacOS/${app.name}"
```

Comment:
Don't use `system` to run the `chmod` executable. Use the standard Tcl `file attributes ... -permissions ...`.